analyzers/src/SonarAnalyzer.CSharp/Extensions/SyntaxNodeAnalysisContextExtensions.cs

```diff
@@ -0,0 +1,32 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using StyleCop.Analyzers.Lightup;
+
+namespace SonarAnalyzer.Extensions
+{
+    public static class SyntaxNodeAnalysisContextExtensions
+    {
+        public static bool IsTopLevelCompilationUnit(this SyntaxNodeAnalysisContext context) =>
```

Comment:
Seeing this again:
I would add this function to the `ISymbolExtentions` so that it takes as an input an `ISymbol symbol` which checks if `symbol.Name.Equals(MainMethodImplicitName )`.

This is because I've understood that `context.ContainingSymbol` checks the symbol for the registered node.
If we decide to check for example the node's parent, this should not properly work. While with the change I proposed above we would give the parent node's containing symbol and it should be fine.

Reply:
The idea was to control it on the context level, since you won't need to remember which kind of property to use for the check within a few months from now. :) Unless @costin-zaharia-sonarsource  is ready to change his mind on that, I propose to leave it as it is now.

Reply:
At a second look at this, I think that Mary's suggestion makes sense: this method does not need the full context to assert that the symbol is the one for top-level functions. Do to this we can move it to `ISymbolEextensions`. With this we could also change the name to something similar with `IsTopLevelMain` since it will only look for that.